ln-resource-mgr/src/forward_manager.rs

```diff
@@ -150,44 +150,66 @@ impl ForwardManagerImpl {
 
         // Check reputation and resources available for the forward.
-        let outgoing_channel = &mut self
-            .channels
-            .get_mut(&forward.outgoing_channel_id)
-            .ok_or(ReputationError::ErrOutgoingNotFound(
-                forward.outgoing_channel_id,
-            ))?
-            .outgoing_direction;
+        let outgoing_channel = self.channels.get_mut(&forward.outgoing_channel_id).ok_or(
+            ReputationError::ErrOutgoingNotFound(forward.outgoing_channel_id),
+        )?;
+
+        // Revenue over the outgoing channel that the incoming channel must meet.
+        let outgoing_threshold = outgoing_channel
+            .bidirectional_revenue
+            .value_at_instant(forward.added_at)?;
+
+        // Reputation from the outgoing channel
+        let outgoing_reputation = outgoing_channel
+            .outgoing_direction
+            .outgoing_reputation(forward.added_at)?;
 
         Ok(AllocationCheck {
             reputation_check: ReputationCheck {
-                outgoing_reputation: outgoing_channel.outgoing_reputation(forward.added_at)?,
-                revenue_threshold,
-                in_flight_total_risk: self.htlcs.channel_in_flight_risk(
-                    ChannelFilter::OutgoingChannel(forward.outgoing_channel_id),
-                ),
-                // The underlying simulation is block height agnostic, and starts its routes with a height of zero, so
-                // we can just use the incoming expiry to reflect "maximum time htlc can be held on channel", because
-                // we're calculating expiry_in_height - 0.
-                htlc_risk: self
-                    .htlcs
-                    .htlc_risk(forward.fee_msat(), forward.expiry_in_height),
+                incoming_reputation: ReputationValues {
+                    reputation: incoming_reputation,
+                    revenue_threshold: outgoing_threshold,
+                    in_flight_total_risk: self.htlcs.channel_in_flight_risk(
+                        ChannelFilter::IncomingChannel(forward.incoming_ref.channel_id),
+                    ),
+                    htlc_risk: self
+                        .htlcs
+                        .htlc_risk(forward.fee_msat(), forward.expiry_out_height),
+                },
+                outgoing_reputation: ReputationValues {
+                    reputation: outgoing_reputation,
+                    revenue_threshold: incoming_revenue_threshold,
+                    in_flight_total_risk: self.htlcs.channel_in_flight_risk(
+                        ChannelFilter::OutgoingChannel(forward.outgoing_channel_id),
+                    ),
+                    // The underlying simulation is block height agnostic, and starts its routes with a height of zero, so
+                    // we can just use the incoming expiry to reflect "maximum time htlc can be held on channel", because
+                    // we're calculating expiry_in_height - 0.
```

Comment:
Let's move this comment up and generalize it to being able to use expiry height (since this applies to `expiry_out_height` above as well)